OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignalCacheCleaner.java

```diff
@@ -4,69 +4,188 @@
-     * 1. In App Messaging id sets (impressions, clicks, views)
-     * 2. Notifications after 1 week
-     * 3. Unique outcome events linked to notification ids (1 week)
+     * 1. Notifications & unique outcome events linked to notification ids (1 week)
+     * 2. Cached In App Messaging Sets in SharedPreferences (impressions, clicks, views) and SQL IAMs
      */
-    synchronized static void cleanOldCachedData(final Context context) {
-        new Thread(new Runnable() {
-            @Override
-            public void run() {
-                Thread.currentThread().setPriority(Process.THREAD_PRIORITY_BACKGROUND);
-                OneSignalDbHelper dbHelper = OneSignalDbHelper.getInstance(context);
-                SQLiteDatabase writableDb = dbHelper.getWritableDbWithRetries();
+    static void cleanOldCachedData(final Context context) {
+        OneSignalDbHelper dbHelper = OneSignalDbHelper.getInstance(context);
+        SQLiteDatabase writableDb = dbHelper.getSQLiteDatabaseWithRetries();
 
-                cleanInAppMessagingCache();
-                cleanNotificationCache(writableDb);
-            }
-        }, OS_DELETE_OLD_CACHED_DATA).start();
-    }
-
-    /**
-     * TODO: Needs to be implemented to clean out old IAM data used to track impressions, clicks, and viewed IAMs
-     */
-    static void cleanInAppMessagingCache() {
-        // NOTE: Currently IAMs will pile up overtime and since IAMs can be modified, active, inactive, etc.
-        //  we never truly know when it is the correct time to remove these ids form our cache
+        cleanNotificationCache(writableDb);
+        cleanCachedInAppMessages(writableDb);
     }
 
     /**
      * Cleans two notification tables
      * 1. NotificationTable.TABLE_NAME
      * 2. CachedUniqueOutcomeNotificationTable.TABLE_NAME
      */
-    static void cleanNotificationCache(SQLiteDatabase writableDb) {
-        cleanOldNotificationData(writableDb);
-        cleanOldUniqueOutcomeEventNotificationsCache(writableDb);
-    }
+    synchronized static void cleanNotificationCache(final SQLiteDatabase writableDb) {
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                Thread.currentThread().setPriority(Process.THREAD_PRIORITY_BACKGROUND);
 
-    /**
-     * Deletes any notifications with created timestamps older than 7 days
-     */
-    private static void cleanOldNotificationData(SQLiteDatabase writableDb) {
-        writableDb.delete(NotificationTable.TABLE_NAME,
-                NotificationTable.COLUMN_NAME_CREATED_TIME + " < " + ((System.currentTimeMillis() / 1_000L) - 604_800L),
-                null);
+                cleanCachedNotifications(writableDb);
+                cleanCachedUniqueOutcomeEventNotifications(writableDb);
+            }
+
+            /**
+             * Deletes notifications with created timestamps older than 7 days
+             */
+            private void cleanCachedNotifications(SQLiteDatabase writableDb) {
```

Comment:
why are these methods (cleanCachedNotifications and cleanCachedUniqueOutcomeEventNotifications) inside the thread definition?